internal/logging/sentry_replacer.go

```diff
@@ -0,0 +1,85 @@
+package logging
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+var filters = []string{
+	// Example (AWS SDK): RequestID: ca767444-d1f9-11ed-afa1-0242ac120002
+	`[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}`,
+	// Example (AWS SDK): arn:aws:iam::4328974392798432:role/my-role-123
+	`arn:aws:[[:word:]]+::\d+:[[:word:]\*-]+/[[:word:]\*-]+`,
+}
+
+var replacement = []byte{'?'}
+
+type SentryReplacer struct {
+	buf    []byte
+	w      io.Writer
+	re     *regexp.Regexp
+	mu     sync.Mutex
+	closed bool
+}
+
+func NewSentryReplacer(w io.Writer) *SentryReplacer {
+	sr := SentryReplacer{
+		w:  w,
+		re: regexp.MustCompile("(" + strings.Join(filters, "|") + ")"),
```

Comment:
This is compiled down to just a single state machine, which is much faster than going over a list of expressions. As we find more and more of these patterns it could get to the point this would be slow.